### ISSUE TITLE
GCC 12.1 support

### ISSUE THREAD

**Issue by tlaurion** [richfelker/musl-cross-make#159]
`-Oz` looks really promising for embedded devices!
> Optimize aggressively for size rather than speed. This may increase the number of instructions executed if those instructions require fewer bytes to encode. -Oz behaves similarly to -Os including enabling most -O2 optimizations.

src:https://gcc.gnu.org/onlinedocs/gcc-12.1.0/gcc/Optimize-Options.html

**Comment by tlaurion** [comment]
And now I'm confused. Is Oz Apple related?

https://gcc.gnu.org/legacy-ml/gcc/2017-08/msg00281.html

**Comment by wdlkmpx** [comment]
Somebody is probably already compiling GCC 12.2 with musl-cross-make, but we have to find the repository that contains DA changes

**Comment by dangerousHobo** [comment]
I just made a build with 12.1 and was able to compile with `-Oz` and run on x86-64.

It's difficult to find history on `-Oz`, but it may be that this was originally for Apple only and more recently expanded to more platforms/architectures.

**Comment by AlexKordic** [comment]
Hi @aka-mj, how did you built with gcc 12.1? The musl.cc download page offers `GCC: 11.2.1`

**Comment by tlaurion** [comment]
@aka-mj ping

**Comment by tlaurion** [comment]
http://port70.net/~nsz/musl/gcc-12.2.0/ ?

**Comment by aka-mj** [comment]
@tlaurion, @AlexKordic, I built this via gcc 12.1 by modifying the Makefile to use GCC 12.1.

- GCC_VER = 9.4.0
+ GCC_VER = 12.1.0

**Comment by tlaurion** [comment]
@aka-mj not sure how that would work
- patches missing
- checksum validation missing

My last post was about patches having been done downstream.

I guess my question is if it would be OK if a or was made with that content even if I'm not the original author, referencing to whare it was found.

**Comment by aka-mj** [comment]
@tlaurion , I didn't add patches but I did add the package checksum.

https://github.com/aka-mj/musl-cross-make/commit/c7b0311a179f167583dce8306b9953edf37d08e2

**Comment by sean-mcmanus** [comment]
We've encountered a gcc 11 crash bug with std::regex_match that is fixed with gcc 12.

**Comment by rofl0r** [comment]
@sean-mcmanus gcc 11 was just updated to 11.4. does the bug still exist there ? if so please provide a link to the bugtracker issue report so we can backport the patch.

**Comment by rofl0r** [comment]
if it's related to stack size, it may well be that musl's default thread stack of 128 (or was it enlarged to 180?) KB isn't sufficient for your application - that is if you encounter this from a thread that's not the main thread. in that case you can request a bigger stack with pthread_attr_setstacksize(3p).

**Comment by sean-mcmanus** [comment]
@rofl0r Yeah, increasing the thread stack size fixed it. Thanks. We had already done that for other scenarios, but this case went over our previous increase.

**Comment by tlaurion** [comment]
@rofl0r may I ask why musl-cross doesn't intent to provide 12.x support?

**Comment by richfelker** [comment]
There's no intent that it not be provided; it's just not something I have tested patches for, and as maintainer I'm not going to merge something for which I have no idea whether it works or (in the case of some of the PRs) that's making other random wrong changes alongside adding support for new versions.

12.x and 13.x are coming at some point, once I get to them. Well-documented PRs that make no extraneous changes and that cite where the patches are from and why any that have been dropped are no longer needed would probably help move this along.
